sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java

```diff
@@ -68,63 +110,122 @@ public void onSpanFinished(final @NotNull ISpan span) {
     if (!enabled) {
       return;
     }
+
     if (span instanceof NoOpSpan) {
       return;
     }
+
     if (span instanceof NoOpTransaction) {
       return;
     }
 
-    @Nullable SentryFrameMetrics diff = null;
+    captureFrameMetrics(span);
+
     synchronized (lock) {
-      final @Nullable SentryFrameMetrics metricsAtStart =
-          metricsAtSpanStart.remove(span.getSpanContext().getSpanId());
-      if (metricsAtStart != null) {
-        diff = currentFrameMetrics.diffTo(metricsAtStart);
+      if (runningSpans.isEmpty()) {
+        clear();
+      } else {
+        // otherwise only remove old/irrelevant frames
+        final SpanStart oldestSpan = runningSpans.first();
+        frames.tailSet(oldestSpan.timeNanos).clear();
```

Comment:
here we lose info in case of multiple spans running at the same time
frame delay should be added to all of them, but we're adding only to one in this case

Reply:
Here `runningSpans` will contain all  running spans (except the one which just finished above). Then we get the "oldest" span which is still running and delete all frame data prior to it.

Reply:
Update: Found the issue and added tests, thanks!